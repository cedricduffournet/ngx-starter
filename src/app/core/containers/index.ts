export * from './toaster';
export * from './http-progress-bar/http-progress-bar.component';
