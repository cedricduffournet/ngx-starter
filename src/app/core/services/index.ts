export * from './http-progress.interceptor';
export * from './http.service';
