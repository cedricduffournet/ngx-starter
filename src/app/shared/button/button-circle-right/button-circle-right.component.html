<div>
  <button
    type="button"
    class="btn-circle btn-xl btn btn-absolute-corner-right"
    ngClass="btn-{{ type }}"
  >
    {{ label }}
  </button>
</div>
