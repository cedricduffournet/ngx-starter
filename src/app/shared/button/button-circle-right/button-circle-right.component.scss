@import 'src/variables.scss';

.btn-circle {
  width: 30px;
  height: 30px;
  text-align: center;
  padding: 6px 0;
  font-size: 12px;
  line-height: 1.428571429;
  border-radius: 15px;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12),
    0 3px 1px -2px rgba(0, 0, 0, 0.2);
  &.btn-xl {
    width: 80px;
    height: 80px;
    padding: 10px 16px;
    font-size: 26px;
    line-height: 1.33;
    border-radius: 40px;
  }
  &.btn-absolute-corner-right {
    position: fixed;
    right: map-get($spacers, 3);
    bottom: map-get($spacers, 3);
  }
}
