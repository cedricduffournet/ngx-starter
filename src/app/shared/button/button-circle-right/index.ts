export * from './button-circle-right.component';
