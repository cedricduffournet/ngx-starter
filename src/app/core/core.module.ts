import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { environment } from '@env/environment';
import {
  HttpClientModule,
  HttpClient,
  HTTP_INTERCEPTORS
} from '@angular/common/http';

import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';
import { ModalModule } from 'ngx-bootstrap/modal';
import { ToasterModule } from 'angular2-toaster';

import {
  ToasterComponent,
  HttpProgressBarComponent
} from '@app/core/containers';
import { HttpProgressInterceptor } from '@app/core/services/http-progress.interceptor';

const MODULES: any[] = [
  HttpClientModule,
  TranslateModule.forRoot({
    loader: {
      provide: TranslateLoader,
      useFactory: HttpLoaderFactory,
      deps: [HttpClient]
    }
  }),
  ModalModule.forRoot(),
  ToasterModule.forRoot(),
  CommonModule
];

export const COMPONENTS = [ToasterComponent, HttpProgressBarComponent];

@NgModule({
  imports: MODULES,
  declarations: COMPONENTS,
  exports: COMPONENTS,
  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpProgressInterceptor,
      multi: true
    }
  ]
})
export class CoreModule {
  constructor(
    @Optional()
    @SkipSelf()
    parentModule: CoreModule
  ) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import only once in main AppModule.'
      );
    }
  }
}

export function HttpLoaderFactory(http: HttpClient) {
  return new TranslateHttpLoader(
    http,
    `${environment.prefixAsset}/assets/i18n/`,
    '.json'
  );
}
