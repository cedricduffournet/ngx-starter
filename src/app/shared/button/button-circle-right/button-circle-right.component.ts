import { Component, Input, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'app-button-circle-right',
  templateUrl: './button-circle-right.component.html',
  styleUrls: ['./button-circle-right.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ButtonCircleRightComponent {
  @Input() label = '';
  @Input() type = 'success';
}
