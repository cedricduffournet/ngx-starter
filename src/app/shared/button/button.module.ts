import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { ButtonCircleRightComponent } from '@app/shared/button/button-circle-right';

@NgModule({
  imports: [CommonModule],
  exports: [ButtonCircleRightComponent],
  declarations: [ButtonCircleRightComponent]
})
export class ButtonModule {}
