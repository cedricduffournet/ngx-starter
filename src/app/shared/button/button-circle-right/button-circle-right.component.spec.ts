import { Component } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ButtonCircleRightComponent } from '@app/shared/button/button-circle-right';

describe('ButtonCircleRightComponent', () => {
  let fixture: ComponentFixture<ButtonCircleRightComponent>;
  let component: ButtonCircleRightComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [ButtonCircleRightComponent]
    });

    fixture = TestBed.createComponent(ButtonCircleRightComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should display label', () => {
    component.label = 'test label';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should display button with danger', () => {
    component.type = 'danger';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
