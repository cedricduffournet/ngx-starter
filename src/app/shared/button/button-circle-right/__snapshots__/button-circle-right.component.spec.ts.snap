// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCircleRightComponent should be created 1`] = `
<app-button-circle-right
  label=""
  type={[Function String]}
>
  <div>
    <button
      class="btn-circle btn-xl btn btn-absolute-corner-right btn-success"
      ng-reflect-klass="btn-circle btn-xl btn btn-abso"
      ng-reflect-ng-class="btn-success"
      type="button"
    >
        
    </button>
  </div>
</app-button-circle-right>
`;

exports[`ButtonCircleRightComponent should display button with danger 1`] = `
<app-button-circle-right
  label=""
  type={[Function String]}
>
  <div>
    <button
      class="btn-circle btn-xl btn btn-absolute-corner-right btn-danger"
      ng-reflect-klass="btn-circle btn-xl btn btn-abso"
      ng-reflect-ng-class="btn-danger"
      type="button"
    >
        
    </button>
  </div>
</app-button-circle-right>
`;

exports[`ButtonCircleRightComponent should display label 1`] = `
<app-button-circle-right
  label={[Function String]}
  type={[Function String]}
>
  <div>
    <button
      class="btn-circle btn-xl btn btn-absolute-corner-right btn-success"
      ng-reflect-klass="btn-circle btn-xl btn btn-abso"
      ng-reflect-ng-class="btn-success"
      type="button"
    >
       test label 
    </button>
  </div>
</app-button-circle-right>
`;
